lib/cn/cn_tree.c

```diff
@@ -2354,6 +2356,84 @@ cn_tree_fini(void)
     }
 }
 
+static int
+kblk_compare(const void *a, const void *b)
+{
+    const struct kvset_kblk *kblk_a = a, *kblk_b = b;
+
+    /* Return the kblock with the largest min key. */
+    return -1 * key_disc_cmp(&kblk_a->kb_kdisc_min, &kblk_b->kb_kdisc_min);
+}
+
+static merr_t
+get_split_key(const struct cn_tree_node *node, void *key_buf, size_t key_buf_sz, uint16_t *key_len)
+{
+    merr_t err;
```

Comment:
callers need to have cn tree read lock before calling this function.  should state that in function comment.

also, the iterators don't need refs on the kvsets since caller has read lock and iterators are run to completion before returning.

Reply:
I'll work on adding a function comment. It is worth one. Will remove the ref management